### ISSUE TITLE
isLockedOut() funtion in Member.php does not check for FailedLoginCount value

### ISSUE THREAD

**Issue by Anytech** [silverstripe/silverstripe-framework#9788]
Silverstripe 3-4+

After changing the FailedLoginCount on a member they are still unable to log in. 

Upon inspecting the lock out checking code there is no checking of this overriding value and the user cannot proceed unless the attempts are removed from the database.

This FailedLoginCount value also needs to be checked and not just the default 10 last attempts which does not clear the lockout.

src/Security/Member.php

```
    /**
     * Returns true if this user is locked out
     *
     * @skipUpgrade
     * @return bool
     */
    public function isLockedOut()
    {
        /** @var DBDatetime $lockedOutUntilObj */
        $lockedOutUntilObj = $this->dbObject('LockedOutUntil');
        if ($lockedOutUntilObj->InFuture()) {
            return true;
        }

        $maxAttempts = $this->config()->get('lock_out_after_incorrect_logins');
        if ($maxAttempts <= 0) {
            return false;
        }

        $attempts = LoginAttempt::getByEmail($this->Email)
            ->sort('Created', 'DESC')
            ->limit($maxAttempts);

        if ($attempts->count() < $maxAttempts) {
            return false;
        }

        foreach ($attempts as $attempt) {
            if ($attempt->Status === 'Success') {
                return false;
            }
        }

        // Calculate effective LockedOutUntil
        /** @var DBDatetime $firstFailureDate */
        $firstFailureDate = $attempts->first()->dbObject('Created');
        $maxAgeSeconds = $this->config()->get('lock_out_delay_mins') * 60;
        $lockedOutUntil = $firstFailureDate->getTimestamp() + $maxAgeSeconds;
        $now = DBDatetime::now()->getTimestamp();
        if ($now < $lockedOutUntil) {
            return true;
        }

        return false;
    }
```

**Comment by Anytech** [comment]
Possibly a button that removes the lockout would suffice.

**Comment by lekoala** [comment]
I had the same issue and my customer wanted to be able to unlock users manually

I've ended up with this which delete failed attempt from the database

https://github.com/lekoala/silverstripe-base/blob/master/src/Security/BaseMemberExtension.php#L320

i don't know if it's the best way to do it, but it works :-)

**Comment by michalkleiner** [comment]
Ha! I wondered why resetting the number in the CMS field didn't help. Good find!

**Comment by sunnysideup** [comment]
In SIlverstripe 5, you need to reset both the failed log-in count and the `LockedOutUntil` field.  This second field will determine how long the user is logged out for and the Failed Login Count will reset to (not zero, but that is another issue).
